src/security-rules/security-rules-api-client.ts

```diff
@@ -49,11 +111,31 @@ export class SecurityRulesApiClient {
    * @param {string} name Full qualified name of the resource to get.
    * @returns {Promise<T>} A promise that fulfills with the resource.
    */
-  public getResource<T>(name: string): Promise<T> {
+  private getResource<T>(name: string): Promise<T> {
     const request: HttpRequestConfig = {
       method: 'GET',
       url: `${this.url}/${name}`,
     };
+    return this.sendRequest<T>(request);
+  }
+
+  private getRulesetName(name: string): Promise<string> {
```

Comment:
does this need to return a `Promise`? it looks purely synchronous

Reply:
This was done so that errors result in rejections as expected. But we could achieve the same by starting the promise chain before calling `getRulesetName`. 

Changed into a sync method.